Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2806,6 +2807,8 @@ - (YapDatabaseConnection *)uiDatabaseConnection
     NSAssert([NSThread isMainThread], @"Must access uiDatabaseConnection on main thread!");
     if (!_uiDatabaseConnection) {
         _uiDatabaseConnection = [self.primaryStorage newDatabaseConnection];
+        // Increase object cache limit. Default is 250.
+        _uiDatabaseConnection.objectCacheLimit = 500;
```

Comment:
From https://github.com/yapstudios/YapDatabase/wiki/Performance-Pro

We've gotten into the habit of creating new databaseConnections all the time. Creating the connection isn't expensive in and of itself now that we're using a keySpec, but when possible we should try to re-use the connection to take advantage of it's object cache.

e.g. in the case of the media gallery, it's very likely that much of the "nearby" media messages/attachments are already in cache.

So two changes in this PR:

1. drive the MediaGallery from the ConversationView's uiDatabaseConnection
2. bump up this objectCacheLimit a bit - we could likely benefit from making it even larger, but this is a start.

Reply:
Good point.

Reply:
We only need to share/re-use connections when doing something perf-intensive (and usually user-facing), right?

Reply:
I think that's right - basically anything where latency is a concern.